src/Core/Workspace/Project.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using NuGet.Frameworks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using System.Xml.XPath;
+
+namespace Microsoft.Quantum.IQSharp
+{
+    internal static class ProjectExtensions
+    {
+        public static string ToLogString(this IEnumerable<Project> projects) =>
+            string.Join(";", projects.Select(p => p.ProjectFile));
+    }
+
+    internal class ProjectFileComparer : EqualityComparer<Project>
+    {
+        public override bool Equals(Project p1, Project p2) => p1.ProjectFile == p2.ProjectFile;
+
+        public override int GetHashCode(Project project) => project.ProjectFile.GetHashCode();
+    }
+
+    public class Project
```

Comment:
I think you added this in the other PR, but in case not, should this have a public doc comment?

Reply:
Good point, yes, I missed this. I'll go add it in the other PR.